src/QMCDrivers/MCPopulation.cpp

```diff
@@ -283,6 +283,9 @@ void MCPopulation::set_variational_parameters(const opt_variables_type& active)
 {
   for (auto it_twfs = walker_trial_wavefunctions_.begin(); it_twfs != walker_trial_wavefunctions_.end(); ++it_twfs)
   {
+    opt_variables_type tmpVar;
+    (*it_twfs).get()->checkInVariables(tmpVar);
+    (*it_twfs).get()->checkOutVariables(active);
```

Comment:
Some state was not maintained correctly by clone? I expect only resetParameters is necessary.
Please also use a range based loop.

Reply:
This seems like nonsense since checkInVariables documentation reads like it should have no significant side effects and its purpose is to populate the aggregate optimizable parameters list. If this must be called with the dummy variableSet because some component with optimizable parameters requires a side effect, file an issue if possible assign it against the author and record what requires this checkInVariables in a comment here.

It seems like this happens in QMCCostFunctionBase::put using the golden twf.
So I'm thinking of this cost function as the Rank scope cost function. And it seems to be the only one.